### DISCUSSION TITLE
When export CSV files column separate error with column value contains CSV  comma separator.

### DISCUSSION THREAD

**Discussion by SSSDNSY** [dbeaver/dbeaver#33255] [Q&A] [answered]
### Description

As the title description and as we known, The CSV file use the COMMA [  ,  ]  to separate different columns , But in some column data value its contains one or more COMMA  ,and the CSV column separate cannot work properly .

Maybe my export setting was error or  dbeaver have other solution for export , why we use CSV,because our user need Microsoft's  OFFICE's Excel.exe to open it ,Without dbeaver CSV export we have no choise.

It's can not be define as a BUG in feature aspect, if you can add new feature: add new export file type like *.xls or *.xlsx .

### DBeaver Version

 24.0.3.202404211624

### Operating System

windows10

### Database and driver

MySQL 8.0.25
MySQL Connector/J mysql-connector-j-8.0.33

### Steps to reproduce

#  DDL
```sql
-- test.test definition

CREATE TABLE `test` (
  `id` varchar(32) NOT NULL COMMENT '主键',
  PRIMARY KEY (`id`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb3 COMMENT='测试表';
```

# DML
```sql
INSERT INTO test (id) VALUES('1,112,1');
INSERT INTO test (id) VALUES('102,3');
INSERT INTO test (id) VALUES('3,1');
INSERT INTO test (id) VALUES('4');
```

# Images of reproduce steps

![QQ截图20240513105220](https://github.com/dbeaver/dbeaver/assets/15624361/eac3b749-eaf1-4b2f-a491-ddfa35dce49a)
![QQ截图20240513105234](https://github.com/dbeaver/dbeaver/assets/15624361/c2f5ff31-e09d-4d80-8266-fe719c064265)
![QQ截图20240513105255](https://github.com/dbeaver/dbeaver/assets/15624361/989d38dc-d4a5-48d4-a770-ad7c97c3aed7)
![QQ截图20240513105319](https://github.com/dbeaver/dbeaver/assets/15624361/1e1dcd2b-65fc-45c3-b26b-dc1b139888c7)
![QQ截图20240513111033](https://github.com/dbeaver/dbeaver/assets/15624361/aa5dcba9-4596-46aa-8b6b-52e6b6a03618)
[test_202405131053.csv](https://github.com/dbeaver/dbeaver/files/15288981/test_202405131053.csv)

### Additional context

Thank you to everyone of dbeaver for their contributions to open source projects, as you see , I'm Chinese and the some Chinese translation in export setting not Chinese , So I want to  do something for this  ,but I have no idea about this, can yours do me a favor or some tips, THANKS again.

**Comment by SSSDNSY** [comment]
about additional context,  I mean some export setting selector  English value not Chinese under this image and so on.
![QQ截图20240513105255](https://github.com/dbeaver/dbeaver/assets/15624361/4b79c57a-bf7c-4a61-a872-d43fdba8f8b9)

**Comment by E1izabeth** [comment, marked as answer]
The software you use to view csv files uses `,` as a columns delimiter. There should be a setting to change it. When I open the result of such export in Excel, I don't experience this issue. You can try opening it with Notepad to see, that here we have just a strings in quotes:
```
"主键"
"1,112,1"
"102,3"
"3,1"
"4"

```

**Comment by SSSDNSY** [comment]
Yes, we can change setting with other columns delimiter, but the column value can contains other columns delimiter too.  the csv file type only use comma delimiter whatever you have quotes in strings, so I'll close this bug issue and still hope it's can be export by the *.xls or *.xlsx file type in the future . 
@E1izabeth

**Comment by humphreygao** [comment]
see https://marketplace.eclipse.org/content/dbeaver-office-integration
